lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java

```diff
@@ -94,6 +98,29 @@ public float compare(float[] v1, float[] v2) {
     public float compare(byte[] v1, byte[] v2) {
       return scaleMaxInnerProductScore(dotProduct(v1, v2));
     }
+  },
+  /**
+   * Binary Hamming distance; Computes how many bits are different in two bytes.
+   *
+   * <p>Only supported for bytes. To convert the distance to a similarity score we normalize using 1
+   * / (1 + hammingDistance)
+   */
+  BINARY_HAMMING_DISTANCE {
+    @Override
+    public float compare(float[] v1, float[] v2) {
+      throw new UnsupportedOperationException(
+          BINARY_HAMMING_DISTANCE.name() + " is only supported for byte vectors");
+    }
+
+    @Override
+    public float compare(byte[] v1, byte[] v2) {
+      return binaryHammingDistance(v1, v2);
+    }
+
+    @Override
+    public Set<VectorEncoding> supportedVectorEncodings() {
+      return EnumSet.of(VectorEncoding.BYTE);
+    }
```

Comment:
```suggestion
    public bool supportsVectorEncoding(VectorEncoding encoding) {
      return encoding == VectorEncoding.BYTE;
    }
```
I am not sure why this returns a set. Do we really do anything with the set except check membership?